src/java/bobbybot/Deadline.java

```diff
@@ -0,0 +1,52 @@
+package bobbybot;
+
+import java.time.LocalDateTime;
+import java.time.format.DateTimeFormatter;
+import java.time.format.DateTimeParseException;
+
+/** Represents a bobbybot. Task that should be completed by a specified Date and Time
+ *  @author mokdarren
+ */
+public class Deadline extends Task{
+    protected LocalDateTime by;
+
+    /**
+     *
+     * @param description description of task
+     * @param by date and time that the task should be completed by
+     */
+    public Deadline(String description, LocalDateTime by)  {
+        super(description);
+        this.by = by;
+    }
+
+    /**
+     * Overloaded constructor used to load task from string
+     * @param description description of task
+     * @param by date and time that the task should be completed by
+     * @param isDone boolean flag if task is done
+     */
+    public Deadline(String description, String by, boolean isDone, DateTimeFormatter formatter) {
```

Comment:
Perhaps the name `by` could be something more descriptive with respect to the context? Maybe something like `doneBy`